legate/core/legion.py

```diff
@@ -904,7 +1019,7 @@ def partition(self, runtime, context, parent, color_space, kind, part_id):
 
 
 class PartitionByWeights(PartitionFunctor):
-    def __init__(self, weights):
+    def __init__(self, weights: Union[FutureMap, Any]) -> None:
```

Comment:
I believe we want `weights` to be always `FutureMap`. Was there a place where an object of type `Any` is actually being passed? If so, I think we should fix it.

Reply:
Line 1050 has a branch:
```
 elif isinstance(self.weights, list):
```
which mypy will flag as unreachable code with just `FutureMap`:
```
legate/core/legion.py: note: In member "partition" of class "PartitionByWeights":
legate/core/legion.py:1050:9: error: Statement is unreachable  [unreachable]
            elif isinstance(self.weights, list):
            ^
```
Should the branch be removed entirely?

Reply:
Also just FYI any instance of `Union[..., Any]` is a placeholder that is meant to be refined.

Reply:
Ah, didn't realize we had that branch as we never really used it. then I guess `Union[FutureMap, list[int]]` would be more appropriate.

Reply:
https://github.com/nv-legate/legate.core/pull/158/commits/3318038b119b2369a00999f79ae6f93dd8a48c28